### DISCUSSION TITLE
How do you clear ROM drive on ZuluSCSI RP2040 board

### DISCUSSION THREAD

**Discussion by UFO-OS9** [ZuluSCSI/ZuluSCSI-firmware#117] [Q&A] [answered]
After flashing the ROM, in a desperate attempt to get this thing to work & despite replacing the image with a blank rom flash The ROM still now takes priority over my main image.
Is there any way to restore this board to "new" i.e.  clear the ROM or do I just throw it in the dustbin & get my life back ?

**Comment by PetteriAimonen** [comment, marked as answer]
If you mean the [RP2040 ROM drive support](https://github.com/ZuluSCSI/ZuluSCSI-firmware#rom-drive-in-microcontroller-flash), it can be cleared this way:

    To disable a previously programmed ROM drive, create empty file called HD0.rom.

This will erase the ROM drive contents and it will no longer get activated.

**Comment by aperezbios** [comment]
@UFO-OS9 you can also disable it via a zuluscsi.ini directive:

[SCSI]
DisableROMDrive = 1

In this case, the contents in the flash itself are preserved, but the ini file directive disables it.

**Reply by aperezbios** [reply]
Before we go any further, we really need you to provide your zululog.txt file. Otherwise, were not going to get anywhere. Assuming the board is defective without evidence won't get us anywhere useful.

**Reply by UFO-OS9** [reply]
Sure..... zululog.txt file attached
[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/10187688/zululog.txt)

**Reply by UFO-OS9** [reply]
Just to clarify... requirements:
I'm seeking to emulate a single scsi drive of only 580Mb in capacity, on ID 6 with termination enabled (set via dip switch on zulu pcb)
The PC is a custom made unit supplied to the automotive trade circa 1996 which uses OS9 (NOT Apple) operating system, it uses Bootstrap v 2.0 dated 1994 .

Since I tried to flash the ROM - then flashed again with a blank file to deactivate  it.
I am now forced to use an .ini file just to override it. in order to get it to read an image file

**Comment by aperezbios** [comment]
We need a _non-truncated_ log file, in its entirety. What you sent strips all context.

**Reply by UFO-OS9** [reply]
I totally understand what you are saying, re truncation, but the only edit is the last line separation . in order that you may understand the steps taken.. with ease
So heres  a log file  with one blank 1Gb image (no ini file)
[zululog.txt](https://github.com/ZuluSCSI/ZuluSCSI-firmware/files/10192750/zululog.txt)

**Reply by UFO-OS9** [reply]
Barister 1 enters the courtroom and speaks - "My Lords, as a mere mortal ...
I present to the court exhibit one:  An actual copy of the original "non-truncated" zululog.txt file that does NOT strip all context"

EXHIBIT ONE:
[10ms] Platform: ZuluSCSI RP2040
[10ms] FW Version: 1.1.5-release Dec  5 2022 23:42:09
[11ms] DIP switch settings: debug log 0, termination 1
[11ms] SCSI termination is enabled
[12ms] Flash chip size: 2048 kB
[14ms] SCSI target/disk mode selected by DIP switch, acting as a SCSI disk
[33ms] SD card detected, FAT32 volume size: 15173 MB
[34ms] SD MID: 0x03, OID: 0x53 0x44
[34ms] SD Name: SL16G
[35ms] SD Date: 6/200?
[35ms] SD Serial: 0x72D5B535
[37ms] Reading configuration from zuluscsi.ini
[38ms] Active configuration:
[42ms] -- SelectionDelay: 255
[47ms] -- EnableSCSI2 is on
[1109ms] Finding HDD images in directory /:
[1110ms] -- Opening /HD6.hda for id:6 lun:0
[2010ms] ---- Image file is contiguous, SD card sectors 41056 to 2138207
[2011ms] ---- Read prefetch enabled: 8192 bytes
[2064ms] -- Platform supports ROM drive up to 1692 kB
[2068ms] ---- ROM drive disabled in ini file, not enabling
[2068ms] SCSI ID:6 BlockSize:512 Type:0 Quirks:0 ImageSize:1048576kB
[2574ms] Initialization complete!
[80301ms] WARNING: Host used command 0x1A which is affected by drive geometry. Current settings are 63 sectors x 255 heads = 16065 but image size of 2097152 sectors is not divisible. This can cause error messages in diagnostics tools.

The Judge examines and scrutinizes the document .....  

The judge speaks.......    "My good man...... it looks exactly the same as the truncated version"

**Comment by aperezbios** [comment]
You might want to consider toning down your hyperbole. It's really not helpful, and is getting you nowhere useful. 

Nowhere have you actually described _what your actual problem is_ in any coherent manner. ZuluSCSI's firmware is actually derived from BlueSCSI, as well as SCSI2SD. 

If you can't state in a single sentence what the actual technical problem you're having is, and actually describe what is not working, we can't help you. You're being abusive to the people who would like to help you out, whether you realize it or not.

**Comment by UFO-OS9** [comment]
aperezbios wrote:
"Nowhere have you actually described what your actual problem is in any coherent manner"

Try this link: created 6 days ago https://github.com/ZuluSCSI/ZuluSCSI-firmware/discussions/112

Your "talk down" relating to my non truncated log file was  nothing more than abusive.  

When I  purchased this board,  all that arrived, was a pcb in a box - no links no notes no nothing !
I just reached out here, for a bit of help... what did I get ? ............ nothing !
I have learned through life.. never to expect anything & this project clearly defines that theory !

My BlueSCSI arrived today & worked ....straight out of the box,  that is 60 pieces sold to my customers next month alone
They could have been zuluscsi's

So next time, you decide to "talk down" to someone... just have a little respect, because two can play those silly games.

I wish you luck,  with your project,  it looked such a promising device !

**Reply by aperezbios** [reply]
You reached out with a bunch of editorializing. Nowhere did you describe _why_ you were attempting to use a ROM drive. I'm glad you've found a device that should meet your needs.

**Reply by UFO-OS9** [reply]
**I wrote....just a few lines above this ....**
"Just to clarify... requirements:
I'm seeking to emulate a single SCSI drive of only 580Mb in capacity, on ID 6 with termination enabled (set via dip switch on zulu pcb)
The PC is a custom made unit supplied to the automotive trade circa 1996 which uses OS9 (NOT Apple) operating system, it uses Bootstrap v 2.0 dated 1994 ."

Is that  not in a coherent manner ????   Do you suffer from selective viewing  ???

Finally...
We have over 2,500, SCSI driven OS9 units out in the U.K. alone,  with the global market place also requiring a bigger solution.
Every single one, will at sometime, have a SCSI drive failure.

With your arrogance ...... you simply blew it.........  all over, an inserted space in a text file !

Just stop & think about it.  this is  a help forum not a dictatorship !

Good Luck !

**Comment by aperezbios** [comment]
The ROM drive feature in the RP2040 is a very new feature, and it's entirely possible you've hit a legitimate bug, but you've repeatedly chosen to editorialize instead of explaining why you were using that feature in the first place, even when directly asked about it. I was simply curious why you felt it would meet your needs. It's worth pointing out that, had you just used the ZuluSCSI RP2040 in the standard manner, it would have worked exactly like the BlueSCSI you subsequently purchased, when you decided you'd hit a dead end. Setting a single variable in an ini file isn't that much of an inconvenience, is it? It's also not clear from your communications here whether or not you had the board working properly, once you'd disabled the ROM drive. You know what you've done, but we don't, which is why we need as much information as possible.

Without knowing who you are (you're anonymous here, I'm not), as well as who you purchased your ZuluSCSI RP2040 from, we cannot possibly offer you the level of support, such as a board exchange, that you seem to expect. Please send us an e-mail and we'll work with you to see what needs to be done to win your trust. We have hands-on experience helping many industrial/business customers with large-scale deployments, both of ZuluSCSI and of SCSI2SD. 

BlueSCSI is a hobbyist product, designed by hobbyists, and is great for casual retrocomputer enthusiasts, but I think you'll eventually discover that they simply aren't produced in volumes suitable for your needs.
